hack/jenkins/test-dockerized.sh

```diff
@@ -21,7 +21,12 @@ set -o xtrace
 
 retry() {
   for i in {1..5}; do
-    "$@" && return 0 || sleep "${i}"
+    if "$@" 
+    then
+        return 0
+    else
+        sleep "${i}"
+    fi
```

Comment:
Why did it change and what did it fix:
```
In ./hack/jenkins/test-dockerized.sh line 24:
    "$@" && return 0 || sleep "${i}"
         ^-- SC2015: Note that A && B || C is not if-then-else. C may run when A is true.

For more information:
  https://www.shellcheck.net/wiki/SC2015 -- Note that A && B || C is not if-t...
```
Example: 
**Problematic code:**
```
[[ $dryrun ]] && echo "Would delete file" || rm file
```
**Correct code:**
```
if [[ $dryrun ]]
then
  echo "Would delete file"
else
  rm file
fi
```